.github/workflows/release.yml

```diff
@@ -0,0 +1,42 @@
+name: npm-package-release
+on:
+  release:
+    types: 
+      - created
+
+jobs:
+  publish-npm:
+    name: Publish to NPM Registry
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Configure Node
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12'
+        registry-url: 'https://registry.npmjs.org'
+    - name: Install dependencies
+      uses: bahmutov/npm-install@v1
+    - name: Publish
+      run: yarn publish --access public
```

Comment:
I'd suggest to run tests first, before doing the release, and same below. You could create another job called `test` and the 2 existing jobs would define `needs: test` to set their dependency to it.